services/QuillLMS/client/app/bundles/Teacher/helpers/salesForms.tsx

```diff
@@ -0,0 +1,173 @@
+  district_name: string,
+  school_premium_count_estimate: number,
+  teacher_premium_count_estimate: number,
+  student_premium_count_estimate: number,
+  submission_type: string,
+  comment?: string
+}
+
+const fetchDefaults = require("fetch-defaults");
+const baseUrl = process.env.DEFAULT_URL;
+const headerHash = {
+  headers: {
+    "Accept": "application/JSON",
+    "Content-Type": "application/json"
+  }
+};
+const apiFetch = fetchDefaults(fetch, baseUrl, headerHash);
+
+export const getSchoolsAndDistricts = async (type: string) => {
+  const url = `/options_for_sales_form?type=${type}`;
+  const response = await apiFetch(url);
+  const { status } = response;
+
+  if(requestFailed(status)) {
+    return { error: 'Failed to fetch school and districts. Please refresh the page.' };
+  } else {
+    const schoolOrDistrictOptions = await response.json();
+    const { options } = schoolOrDistrictOptions;
+    return { options: options };
+  }
+}
+
+export const submitSalesForm = async (salesFormSubmission: SalesFormSubmission) => {
+  const response = await apiFetch('/submit_sales_form', {
+    method: 'POST',
+    body: JSON.stringify({ sales_form_submission: salesFormSubmission })
+  });
+  const { status } = response;
+
+  if(requestFailed(status)) {
+    return { error: 'Failed to submit form. Please try again.' };
+  } else {
+    return { success: true };
+  }
+}
+
+export function schoolSearch(options) {
+  const fuse = new Fuse(options, {
+    keys: ['name', 'groupName', 'items.name'],
+    threshold: 0.3,
+  });
+
+  return (value) => {
+    if (!value.length) {
+      return options;
+    }
+    const results = fuse.search(value)
+    if(value && !results.length) {
+      return [{ name: SCHOOL_NOT_LISTED, value: SCHOOL_NOT_LISTED}];
+    }
+    return fuse.search(value);
+  };
+}
+
+export const districtSearch = (options) => {
+  const fuse = new Fuse(options, {
+    keys: ['name', 'groupName', 'items.name'],
+    threshold: 0.3,
+  });
+
+  return (value) => {
+    if (!value.length) {
+      return options;
+    }
+    const results = fuse.search(value)
+    if(value && !results.length) {
+      return [{ name: DISTRICT_NOT_LISTED, value: DISTRICT_NOT_LISTED}];
+    }
+    return fuse.search(value);
+  };
+}
+
+export const renderSchoolAndDistrictSelect = ({
+  errors,
+  schoolNotListed,
+  districtNotListed,
+  selectedSchool,
+  selectedDistrict,
+  schools,
+  districts,
+  handleUpdateField,
+  handleSchoolSearchChange,
+  handleDistrictSearchChange
+}) => {
+  const schoolSearchInput = (
+    <SelectSearch
+      filterOptions={schoolSearch}
+      onChange={handleSchoolSearchChange}
+      options={schools}
+      placeholder="Search for your school"
+      search={true}
+    />
+  );
+  const districtSearchInput = (
+    <SelectSearch
+      filterOptions={districtSearch}
+      onChange={handleDistrictSearchChange}
+      options={districts}
+      placeholder="Search for your district"
+      search={true}
+    />
+  );
```

Comment:
For consistent styling, we should probably use our custom `DropdownInput` backpack element rather than the default `SelectSearch` here. You can see an example of the searchable `DropdownInput` here under "Exposed Editable Dropdown Menu": https://www.quill.org/backpack#menus.

Reply:
the reason I opted to use `SelectSearch` was because of [this part of the spec](https://www.notion.so/quill/Create-custom-sales-forms-8e23e452b9894d5abeed63af7323f41c#69f29dba85074c4184b570e5acc5c429). I can pass the custom search functions I've added in this PR to handle this case. Is there a straightforward way to do this with the `DropdownInput` ?